hypershift-operator/controllers/hostedcluster/internal/platform/powervs/ccm.go

```diff
@@ -0,0 +1,302 @@
+package powervs
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"strings"
+	"text/template"
+
+	ibmcore "github.com/IBM/go-sdk-core/v5/core"
+	"github.com/IBM/platform-services-go-sdk/iamidentityv1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	"github.com/openshift/hypershift/support/upsert"
+)
+
+const (
+	// ConfigMap
+	configMapName         = "cloud-conf"
+	kubeConfigKeyInSecret = "kubeconfig"
+
+	// Deployment
+	deploymentName         = "ibm-cloud-controller-manager"
+	containerName          = "cloud-controller-manager"
+	kubeConfigFileBasePath = "/etc/kubernetes"
+	secretMountPath        = "/etc/vpc"
+	configMapMountPath     = "/etc/ibm"
+)
+
+type configMap struct {
```

Comment:
```suggestion
type ccmConfigTemplateParameters struct {
```

It might be easier to simply not define this type and use a `map[string]string` instead

Reply:
Used `map[string]string{}`